src/app/approve/approve.component.ts

```diff
@@ -301,9 +359,21 @@ export class ApproveComponent implements OnInit {
   }
 
   nanosToUnitString(nanos: number): string {
+    return this.toFixedLengthDecimalString(nanos / 1e9);
+  }
+
+  hexScaledExchangeRateToFloat(hex: Buffer): number {
+    return parseInt(hex.toString('hex'), 16) / 1e38;
```

Comment:
doesn't this get crazy with 1e38? I remember we had to use BigInts to format nanos like in /lib/bindata/util.ts and that already was overflowing hard.

Reply:
Yeah we should use the BN library for this parsing

Reply:
Good point. Looks like the javascript number type [supports values up to ~1e308](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_VALUE), but starts to lose precision [beyond ~9e15](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger)

Guess, the precision loss was just not noticeable in my testing. I'll switch to bigints.